python/bind_slice_methods.h

```diff
@@ -82,8 +83,43 @@ template <class T> struct slicer {
     return self.slice(Slice(dim, i));
   }
 
+  static auto get_by_value(T &self,
+                           const std::tuple<Dim, VariableConstView> &value) {
+    auto [dim, i] = value;
+    return slice(self, dim, i);
+  }
+
   static auto get_range(T &self,
                         const std::tuple<Dim, const py::slice> &index) {
+    auto [dim, py_slice] = index;
+    if constexpr (std::is_same_v<T, DataArray> ||
+                  std::is_same_v<T, DataArrayView>) {
+      try {
+        auto step = py::getattr(py_slice, "step");
+        if (step.is_none()) {
+          std::runtime_error(
+              "Step cannot be specified for value based slicing.");
+        }
+        auto start = py::getattr(py_slice, "start");
+        auto stop = py::getattr(py_slice, "stop");
+        auto start_var =
+            start.is_none()
+                ? Variable{}
```

Comment:
Should this (and below) be `VariableConstView{}`? Otherwise we have a temporary variable that does out of scope, implicitly converted to (and referenced by) a view?

Reply:
This seems to have worked. Thanks a lot.